brave/brave-core
# Import cookies from Chrome

Support importing cookies from Chrome, per https://github.com/brave/brave-browser/issues/47.

I tried to strike a balance of minimizing Chromium patches while avoiding copy-pasting large chunks of Chromium code, which is what the Muon implementation does to keep patching to an absolute minimum. As a reviewer, if you see any places where patching could be reduced without sacrificing maintainability, please let me know!

## Submitter Checklist:


## Test Plan:

### Automated testing

1. `yarn test brave_unit_tests`
    - Output should contain `ChromeImporterTest.ImportCookies` and `SUCCESS: all tests passed.`

### Manual testing

1. You will need a Chrome profile with some cookies saved in it. I have been using a test profile where I log in to an account on a specific website, e.g. https://asana.com, so the auth cookies will be saved in the Chrome profile.
2. Use a clean Brave profile: `rm -rf ~/Library/Application\ Support/Brave-Development/`
3. `yarn start`
4. In Brave, navigate to your test website, e.g. https://asana.com. You should _not_ be logged in, because you haven't imported any cookies yet.
5. Open the main menu (e.g. **Brave** on macOS) and choose **Import Bookmarks and Settings...**
6. Choose your test Chrome profile from the dropdown menu of browser profiles.
7. Make sure the **Cookies** checkbox is activated.
8. Click **Import**.
9. Depending on your platform, you may be prompted several times to unlock keychain entries **Chrome Safe Storage** or **Brave Safe Storage**. If you are, enter your password and click **Allow** or **Always allow**.
10. Click **Done** to close the import dialog.
11. Refresh the page in your original tab (e.g. https://asana.com). You should be logged in.

### Cross-platform testing

I will verify that this PR builds and passes both manual and automated tests on our 3 target platforms: macOS, Windows, and Linux, updating the table below as I go.

|                 | Builds | Manual Tests Pass | Automated Tests Pass |
|----------|---|---|---|
| macOS    | ✅ | ✅ | ✅ |
| Windows | ✅ | ✅ | ❌ |
| Linux       | ✅ | ✅ | ❌ | 

## Reviewer Checklist:
